### ISSUE TITLE
[Feature Request]: Add an option for setting `charset`

### ISSUE THREAD

**Issue by jw-foss** [egoist/rollup-plugin-esbuild#267]
### Background:

I am using this plugin to compile one of my locale file which contains `utf-8` charset instead of simple ascii word.

### Example

Say we have this file:

`locale.zh_CN.ts`
```ts

export default {
  picker: {
    confirm: "确定",
    clear: "清除",
  },
  // ...
}
```

If I run the compiler with options:

`build.js`
```js

const path = require('path')
const { nodeResolve } = require('@rollup/plugin-node-resolve')
const rollup = require('rollup')
// const typescript = require('rollup-plugin-typescript2')
const esbuild = require('rollup-plugin-esbuild')

const root = path.resolve(__dirname, '..');
const file = process.argv[2];
const defaultOpts = {
  input: path.resolve(root, file),
  plugins: [
    nodeResolve(),
    esbuild()
  ],
  external() {
    return true
  },
}

const run = async (name) => {
  const esm = {
    format: 'es',
    file: `es/${name}`,
  };
  const cjs = {
    format: 'cjs',
    file: `lib/${name}`,
    exports: 'named',
  }

  const bundle = await rollup.rollup(defaultOpts);
  await Promise.all([bundle.write(esm), bundle.write(cjs)]);
  console.log(name, 'build finished');
}

run(`locale.zh_CN.ts`)

```

Which produces:

```js
export default {
  colorpicker: {
    confirm: "\u786E\u5B9A",
    clear: "\u6E05\u7A7A"
  }
}

```

### Suggestion
Per [Charset for esbuild](https://esbuild.github.io/api/#charset), `esbuild` allows user to pass an option called `charset`.
Should we add an option for users to pass `charset` to `esbuild` ?

**Comment by leopiccionia** [comment]
I've created a PR, because I also need this feature.

My use case is that my code includes a HTML entities parser, and escaping all those Unicode characters adds several KBs to my bundle.

**Comment by sxzz** [comment]
Resolved in #330
